encoding/protobuf.go

```diff
@@ -106,7 +106,19 @@ func (p protoSerializer) Encoding() Encoding {
 	return Protobuf
 }
 
+func (p protoSerializer) IsClientStreaming() bool {
+	return p.method.IsClientStreaming()
+}
+
+func (p protoSerializer) IsServerStreaming() bool {
+	return p.method.IsServerStreaming()
+}
+
 func (p protoSerializer) Request(body []byte) (*transport.Request, error) {
+	if p.MethodType() != Unary {
+		return nil, fmt.Errorf("request method must not be invoked for a streaming rpc method: %q", p.method.GetInputType().GetFullyQualifiedName())
```

Comment:
nit: the error message should be that request can only be used with unary method. (this error message would become invalid if another type would be added)